stream_alert/rule_processor/handler.py

```diff
@@ -167,68 +174,138 @@ def get_alerts(self):
         """
         return self._alerts
 
+    @staticmethod
+    def _segment_records_by_count(record_list, max_count):
+        """Segment records by length
+
+        Args:
+            record_list (list): The original records list to be segmented
+            max_count (int): The max amount of records to yield per group
+        """
+        for index in range(0, len(record_list), max_count):
+            yield record_list[index:index + max_count]
+
+    def _segment_records_by_size(self, record_batch):
+        """Segment record groups by size
+
+        Args:
+            record_batch (list): The original record batch to measure and segment
+
+        Returns:
+            generator: Used to iterate on each newly segmented group
+        """
+        split_factor = 1
+        len_batch = len(record_batch)
+
+        # Sample the first batch of records to determine the split factor.
+        # Generally, it's very rare for a group of records to have
+        # drastically different sizes in a single Lambda invocation.
+        while len(str(record_batch[:len_batch / split_factor])) > MAX_BATCH_SIZE:
```

Comment:
This has the same length-check discrepancy as line 216. Do you want `len(json.dumps(...))` instead of `len(str(...))`? Up to you

Reply:
Good point, i'll make that change